logstash-core/src/main/java/org/logstash/plugins/discovery/PluginRegistry.java

```diff
@@ -77,6 +115,29 @@ private static void discoverPlugins() {
                 }
             }
         }
+
+        // after loaded all plugins, check if aliases has to be provided
+        addAliasedPlugins(FILTERS);
+        addAliasedPlugins(OUTPUTS);
+        addAliasedPlugins(INPUTS);
+        addAliasedPlugins(CODECS);
+    }
+
+    private static <T> void addAliasedPlugins(Map<String, Class<T>> pluginCache) {
```

Comment:
Note for review.
Here I had the doubt to use streams, introducing a support class:
```java

private static final class NameAndClass<T> {
    final String name;
    final Class<T> clazz;

    NameAndClass(String name, Class<T> clazz) {
        this.name = name;
        this.clazz = clazz;
    }

    @SuppressWarnings("unchecked")
    static <T> NameAndClass<T> fromEntry(Map.Entry<String, T> e) {
        return new NameAndClass<T>(e.getKey(), (Class<T>) e.getValue());
    }

    @SuppressWarnings("unchecked")
    public <T> NameAndClass<T> withName(String newName) {
        return new NameAndClass<T>(newName, (Class<T>) clazz);
    }
}


final Map<String, Class<T>> aliasedToAdd = pluginCache.entrySet().stream()
        .map(NameAndClass::fromEntry)
        .filter(e -> isAliased(e.name))
        .map(n -> n.withName(aliasFromOriginal(n.name)))
        .filter(n -> !pluginCache.containsKey(n.name))
        .collect(Collectors.<NameAndClass, String, Class<T>>toMap(n -> n.name, n -> n.clazz));
```

But we have to create a support class only for this and has a lot of object creation due to stream